lib/rouge/lexers/eex.rb

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class EEX < TemplateLexer
+      title "EEX"
+      desc "Embedded Elixir"
+
+      tag 'eex'
+
+      filenames '*.eex'
+
+      def initialize(opts={})
+        @elixir_lexer = Elixir.new(opts)
+
+        super(opts)
+      end
+
+      start do
+        parent.reset!
+        @elixir_lexer.reset!
+      end
+
+      open  = /<%%|<%=|<%#|<%/
+      close = /%%>|%>/
+
+      state :root do
+        rule %r/<%#/, Comment, :comment
+
+        rule open, Comment::Preproc, :elixir
+
+        rule %r/.+?(?=#{open})|.+/m do
+          delegate parent
+        end
+      end
+
+      state :comment do
+        rule close, Comment, :pop!
+        rule %r/./, Comment
```

Comment:
I don't follow why it was necessary to change this. I checked out your fork repo, @julp, and when I revert this to the pattern that's used by the ERB lexer it lexes the single-line comment in your visual sample fine. Can you give a specific example that doesn't work?

Reply:
On:
```
<%# xxx %>
foo
<%# xxx %>
bar
```
With `rule /.+(?=#{close})|.+/m, Comment`, foo becomes part of the comment but after some digging the problem is simply that `.+` is greedy (should be `.+?`) so `%>` is "miss detected"/consumed and I just saw that the same problem was fixed since on the erb lexer (issue 900 ; commit 3fe2359)

Reply:
Ah, yeah. That makes sense :)